LibertyDSNP/example-client
# view a thumbnail for all images

Purpose
---------------
Feature
[https://www.pivotaltracker.com/story/show/179954697](https://www.pivotaltracker.com/story/show/179954697)
As a user, I should be able to view a thumbnail for images such as:
[https://placekitten.com/300/300](https://placekitten.com/300/300)


Solution
---------------
- Currently, if there is no .png/.gif/.jpg/etc. in the extension, it is returning undefined and displaying the alt image instead of the actual valid image in the thumbnail.
- make sure that valid image url that does not have an extension is used to create the thumbnail.
- Add this case to tests.


Change summary
---------------
* create helper function to return thumbnail url if there is a valid case
*  onError, return empty string, causing the component to display the alternative image
* add test case


Steps to Verify
----------------
1. add youtube video
2. add vimeo
3. add image with no extension
4. add image with extension
5. add mp4, audio, or other edge case
6. make sure they all work and display the expected


Additional details / screenshot
----------------
links in order from left to right
 - https://vimeo.com/618751631
 - https://www.youtube.com/watch?v=tHeEzJfGQu4
 - https://placekitten.com/300/300
 - https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg
 - https://www.learningcontainer.com/wp-content/uploads/2020/05/sample-mp4-file.mp4

https://user-images.githubusercontent.com/43625033/138187453-3adc31cf-0bb3-4f7a-b910-997ca5bdfeef.mov
